### ISSUE TITLE
Running Peergos with custom native-lib path still  creates native-lib in local path

### ISSUE THREAD

**Issue by raspher** [Peergos/web-ui#540]
Discovered while packaging peergos on NixOs. When i run `java -Djava.library.path=result/share/peergos/native-lib/ -jar result/share/java/peergos.jar` it creates **native-lib/libtweetnacl.so** in current `pwd`.

```shell
#: java -Djava.library.path=result/share/peergos/native-lib -jar result/share/java/peergos.jar 
Jan 22, 2024 10:11:34 AM peergos.server.util.Logging init
INFO: Logging to /home/raspher/.peergos/peergos.0.log
Run with -help to show options
```

What i'm doing:
1. get `https://github.com/Peergos/web-ui/releases/download/v${version}/Peergos.jar` to `$out/share/java/`
2. `mkdir -p $out/share/peergos && cd $out/share/peergos`
3. run peergos to generate `native-libs` in this directory for current architecture
4. create a wrapper (by using `$out/bin/peergos` user can pass arguments without typing `java ...`)

As you may saw, running `java -Djava.library.path=result/share/peergos/native-lib -jar result/share/java/peergos.jar` does not spam about `To use the native linux-x86-64 crypto implementation use option -Djava.library.path=native-lib` but it still creates `native-lib` in `pwd` instead of just using provided `java.library.path`

Related work can be found here: https://github.com/NixOS/nixpkgs/pull/282470

**Comment by raspher** [comment]
Well, it's not problem with specific OS but rather problem with user experience and consequences. If you run peergos in multiple places (no matter why, today it was ~/Downloads, yesturday ~/Desktop) you'll have multiple places with created `native-lib` directory.

If you must generate any folder or files, there are places to create them:
 - linux: do it inside `~/.local/share/peergos` 
 - windows: `%appdata%/somewhere_here_idk`
 - macos: idk, never have had

The best option for all use-cases is provide path for `native-lib` AND if not found there create it there. If path is not specified, go by default to OS-specific locations previously mentioned.

**Comment by ianopolous** [comment]
The file is 32kb so shouldn't cause anyone any problems, even with 1000 copies of it. Peergos creates a bunch of other files in the ~/.peergos dir by default (mainly sqlite files), though this can be controlled with the -PEERGOS_PATH arg or env var.

We only use native-lib on x86 + linux. No other arch is supported, but everything still works fine.

**Comment by raspher** [comment]
Yeah, it works i am not saying it is not working. But thrust me, if you run `peergos` inside your home directory and it poops a folder there. Don't tell me you'd like to have for each application you have 1 poopped folder on your desktop. Especially when you can run it in multiple places...

**Comment by ianopolous** [comment]
Yep, people can control the .peergos dir with the arg above.

**Comment by ianopolous** [comment]
the native-lib is the only thing not in that dir we create, but it is an arg to the JVM, not to peergos. So we don't actually get the arg in peergos.

**Comment by raspher** [comment]
So how it's created?

**Comment by ianopolous** [comment]
It's basically tweetnacl.c.

**Comment by raspher** [comment]
I am not asking what is that file or what is inside this folder.

Issue here is that it's created in current `pwd`. Better place for this file would be `~/.peergos` or even `/tmp`.

**Comment by ianopolous** [comment]
This is a philosophical question of whether you want to separate data from code. This lib is code and so really belongs where the jar is. At least that's why it is where it is now. I guess we could put it in .peergos, I don't feel strongly either way.

**Comment by raspher** [comment]
Native-libs is not created inside the jar path, but it is created in the current directory. If you call `java -jar /usr/opt/peergos/peergos.jar` from `~/Downloads`, then it is created in `~/Downloads`.

It would be nice if it creates `native-libs` inside jars path (in this example `/usr/opt/peergos`, but keep in mind different distributions have different standards for file locations, so it cannot be hard coded)

**Comment by ianopolous** [comment]
Yep we assume you run the jar from the current dir.

I'll see what I can do.

**Comment by rapenne-s** [comment]
This is also an issue in the container version in which we can only add parameters after `java -jar Peergos.jar`, so we can't inject the parameters to use native-libs without having them being interpreted as Peergos parameters.

**Comment by ianopolous** [comment]
This is fixed in the latest master. We use a tmp path, or a supplied path via -native-tweetnacl-path (which can be read-only if the file already exists there, so nix could just separate it out). Thank you for your patience.
